libkbfs/block_journal.go

```diff
@@ -952,70 +967,88 @@ func flushBlockEntries(ctx context.Context, log logger.Logger,
 	return nil
 }
 
-func (j *blockJournal) removeFlushedEntry(ctx context.Context,
-	ordinal journalOrdinal, entry blockJournalEntry) (
-	flushedBytes int64, err error) {
-	// Fix up the block byte count if we've finished a Put.
-	if entry.Op == blockPutOp && !entry.Ignore {
-		id, _, err := entry.getSingleContext()
-		if err != nil {
-			return 0, err
-		}
-		flushedBytes, err = j.getDataSize(id)
-		if err != nil {
-			return 0, err
-		}
-
-		if flushedBytes > j.unflushedBytes {
-			return 0, fmt.Errorf("Block %v is bigger than our current count "+
-				"of journal block bytes (%d > %d)", id,
-				flushedBytes, j.unflushedBytes)
-		}
-		j.unflushedBytes -= flushedBytes
-	}
-
-	earliestOrdinal, err := j.j.readEarliestOrdinal()
+func (j *blockJournal) removeFlushedEntryFromDisk(ordinal journalOrdinal,
```

Comment:
do you need to pass in `removeData` and `dj`? It seems that you could just check in the function whether saveUntilMDFlush is non-nil